### ISSUE TITLE
Support Obsidian style [[#Heading]] Links

### ISSUE THREAD

**Issue by ChildishhAlbino** [landakram/remark-wiki-link#34]
## Context

If have a page:

```markdown
// example.md
# Example Page

## Heading 2
This is some context

### Heading 3
Reference to [[#Heading 2]] // this doesn't work
Another Reference to [[Example#Heading 2]] // this works
```
We get a bit of weird behaviour.

The first heading reference; `[[#Heading 2]]`, returns a WikiLink to 
```html
<a class="new" href="/wiki/#heading-2/">#Heading 2</a>
```

The first heading reference; `[[#Heading 2]]`, returns a WikiLink to 
```html
<a class="new" href="/wiki/example#heading-2/">Example#Heading 2</a>
```

Is this something we could target to solve? I'm happy to give it a crack - I imagine both `mdast-util-wiki-link` and this project would need to change to support it?

**Comment by ChildishhAlbino** [comment]
Apologies - was very sleep deprived when I raised the issue originally - have updated with clearer details and scope.

**Comment by landakram** [comment]
The expectation is that `[[#Heading 2]]` renders to `<a class="new" href="/wiki/example#heading-2/">Example#Heading 2</a>`?

**Comment by ChildishhAlbino** [comment]
@landakram yep, Obsidian treats those two examples as equal. The `[[#Heading 2]]` is effectively a short hand for `[[Example#Heading 2]]`

**Comment by Jsmithrud37** [comment]
Any update on this? I'm interested in using this project for some Obsidian-related tooling, and this is a must have.

**Comment by therdas** [comment]
Hey all! Taking a stab at this - I also need this very same thing

**Comment by manunamz** [comment]
Coming soon to [remark-wikirefs](https://github.com/wikibonsai/remark-wikirefs/) if anyone is still interested.
